src/ILCompiler.RyuJit/src/JitInterface/CorInfoImpl.RyuJit.cs

```diff
@@ -867,5 +887,61 @@ private ISymbolNode GetGenericLookupHelper(CORINFO_RUNTIME_LOOKUP_KIND runtimeLo
             return _compilation.NodeFactory.ReadyToRunHelperFromDictionaryLookup(helperId, helperArgument, MethodBeingCompiled);
         }
 
+        private CorInfoHelpFunc getCastingHelper(ref CORINFO_RESOLVED_TOKEN pResolvedToken, bool fThrowing)
+        {
+            TypeDesc type = HandleToObject(pResolvedToken.hClass);
```

Comment:
I do not think this will work for CPAOT just yet.

- The casting helpers in CoreCLR are not uniform as in CoreRT, e.g. `CORINFO_HELP_ISINSTANCEOFINTERFACE` can be only used for non-variant interfaces only.
- CoreCLR will not recognize the helper IDs that you are adding. R2R images that will end up using these helper will fail to run.

I think this should keep doing what it was doing for R2R.